airflow/www/views.py

```diff
@@ -3547,6 +3553,109 @@ def dataset_dependencies(self):
             {'Content-Type': 'application/json; charset=utf-8'},
         )
 
+    # @format_parameters({'limit': check_limit})
+    @expose('/object/list_datasets')
+    @auth.has_access([(permissions.ACTION_CAN_READ, permissions.RESOURCE_DATASET)])
+    def get_datasets(self):
+        """Get datasets"""
+        allowed_attrs = ['uri', 'last_dataset_update']
+
+        limit = int(request.args.get("limit", 25))
+        offset = int(request.args.get("offset", 0))
+        order_by = request.args.get("order_by", "uri")
+        lstripped_orderby = order_by.lstrip('-')
+
+        if lstripped_orderby not in allowed_attrs:
+            return {
+                "detail": (
+                    f"Ordering with '{lstripped_orderby}' is disallowed or the attribute does not "
+                    "exist on the model"
+                )
+            }, 400
+
+        limit = 50 if limit > 50 else limit
+
+        with create_session() as session:
+            if lstripped_orderby == "uri":
+                if order_by[0] == "-":
+                    order_by = (DatasetModel.uri.desc(),)
+                else:
+                    order_by = (DatasetModel.uri.asc(),)
+            elif lstripped_orderby == "last_dataset_update":
```

Comment:
How does one actually hit this block via the UI?

Reply:
The [default order](https://github.com/apache/airflow/pull/26358/files#diff-5ed4b421389b931608cae97949c8e129aa9d4076e52d71a9e7f1b42ff49f77a5R75) is `-last_dataset_update`, but we do not yet have a way for users to explicitly sort this way in the UI.

I can attempt to add this in, or we can add it in later.

Reply:
Part of why I asked is everything following is pretty duplicative for asc/desc. An easy "fix" was to just not have it :)

Reply:
It's absolutely disappointingly messy, that's for sure. 😕

I do think that supporting both asc/desc is useful for answering different questions, but I'm not 100% sure if the nulls first/last for the respective ordering is the most optimal, especially for deployments with a great many datasets. Totally willing to nix and fix the pairings if we don't like that.

Reply:
If we do the categorizing and sorting in the UI a la [this excellent suggestion](https://github.com/apache/airflow/issues/26381#issuecomment-1248227417), we can avoid doing the messy ordering in the SQLA query altogether. 🤔

Reply:
We may want to sort in a couple ways right? By name and by last updated at? I think having the UI just default to desc was so that the latest changes are what a user sees first.

Reply:
Oh also, a blank last updated should probably display after datasets with updates instead of before.